wagon.py

```diff
@@ -26,6 +25,7 @@
 import logging
 import platform
 import tempfile
+import StringIO
```

Comment:
to make this python3 compatible you need to try/except ImportError. the package changed in python 3